### PR TITLE
adding environment variable parameter store for now

### PR DESCRIPTION
**Pull request by ben851**
# Summary | Résumé

The dev api lambda was failing due to the parameter store ENVIRONMENT_VARIABLES being missing. Turns out this was not managed by terraform.

This is a dirty hack for now to get it to work, but we should revisit how we sync environment variables between k8s and lambdas. 

## Related Issues | Cartes liées

* https://app.zenhub.com/workspaces/notify-planning-core-6411dfb7c95fb80014e0cab0/issues/gh/cds-snc/notification-planning-core/315

## Before merging this PR

Read code suggestions left by the
[cds-ai-codereviewer](https://github.com/cds-snc/cds-ai-codereviewer/) bot. Address
valid suggestions and shortly write down reasons to not address others. To help
with the classification of the comments, please use these reactions on each of the
comments made by the AI review:

| Classification      | Reaction | Emoticon |
|---------------------|----------|----------|
| Useful              | +1       | 👍        |
| Noisy               | eyes     | 👀        |
| Hallucination       | confused | 😕        |
| Wrong but teachable | rocket   | 🚀        |
| Wrong and incorrect | -1       | 👎        |

The classifications will be extracted and summarized into an analysis of how helpful
or not the AI code review really is.

## Test instructions | Instructions pour tester la modification

TF Apply on staging should show no changes (only create this file on bootstrap)
TF apply on staging may show changes to outputs

## Release Instructions | Instructions pour le déploiement

None.

## Reviewer checklist | Liste de vérification du réviseur

* [ ] This PR does not break existing functionality.
* [ ] This PR does not violate GCNotify's privacy policies.
* [ ] This PR does not raise new security concerns. Refer to our GC Notify Risk Register document on our Google drive.
* [ ] This PR does not significantly alter performance.
* [ ] Additional required documentation resulting of these changes is covered (such as the README, setup instructions, a related ADR or the technical documentation).

> ⚠ If boxes cannot be checked off before merging the PR, they should be moved to the "Release Instructions" section with appropriate steps required to verify before release. For example, changes to celery code may require tests on staging to verify that performance has not been affected.

### CODE REVIEW

File: env/sandbox/lambda-api/terragrunt.hcl [hunk 1]
```diff
@@ -65,8 +65,21 @@ dependency "dns" {
 
 dependency "rds" {
   config_path = "../rds"
+  mock_outputs_allowed_terraform_commands = ["validate", "plan"]
```
[the remark this answers is not in the document]
Comment (ben851) [reply env/sandbox/lambda-api/terragrunt.hcl:68]: wut

File: env/sandbox/lambda-api/terragrunt.hcl [hunk 2]
```diff
@@ -65,8 +65,21 @@
...
+  mock_outputs = {
+    database_read_only_proxy_endpoint = "thisisamockstring_database_read_only_proxy_endpoint"
+    database_read_write_proxy_endpoint = "thisisamockstring_database_read_write_proxy_endpoint"
```
[the remark this answers is not in the document]
Comment (ben851) [reply env/sandbox/lambda-api/terragrunt.hcl:71]: wut

File: env/sandbox/lambda-api/terragrunt.hcl [hunk 3]
```diff
@@ -65,8 +65,21 @@
...
+    postgres_cluster_endpoint = "thisisamockstring_postgres_cluster_endpoint"
+  }
 }
 
+dependency "elasticache" {
+  config_path = "../elasticache"
+  mock_outputs_allowed_terraform_commands = ["validate", "plan"]
+  mock_outputs = {
+    redis_primary_endpoint_address = "thisisamockstring_redis_primary_endpoint_address"
```
[the remark this answers is not in the document]
Comment (ben851) [reply env/sandbox/lambda-api/terragrunt.hcl:80]: wut

File: aws/lambda-api/secrets_manager.tf [hunk 4]
```diff
@@ -8,3 +8,122 @@
...
   secret_string = var.new_relic_license_key
 }
+
+resource "aws_ssm_parameter" "environment_variables" {
+  count       = var.bootstrap ? 1 : 0
+  name        = "ENVIRONMENT_VARIABLES"
+  description = "Environment variables for the API Lambda function"
+  type        = "SecureString"
+  tier        = "Advanced"
+  value       = <<EOF
```
[the remark this answers is not in the document]
Comment (ben851) [reply aws/lambda-api/secrets_manager.tf:18]: Would that I could.

File: aws/lambda-api/secrets_manager.tf [hunk 5]
```diff
@@ -8,3 +8,122 @@
   secret_id     = aws_secretsmanager_secret.new-relic-license-key.id
   secret_string = var.new_relic_license_key
 }
+
+resource "aws_ssm_parameter" "environment_variables" {
+  count       = var.bootstrap ? 1 : 0
+  name        = "ENVIRONMENT_VARIABLES"
```
[the remark this answers is not in the document]
Comment (ben851) [reply aws/lambda-api/secrets_manager.tf:14]: You're very generic

File: aws/lambda-api/secrets_manager.tf [hunk 6]
```diff
@@ -8,3 +8,122 @@ resource "aws_secretsmanager_secret_version" "new-relic-license-key" {
   secret_id     = aws_secretsmanager_secret.new-relic-license-key.id
   secret_string = var.new_relic_license_key
 }
+
+resource "aws_ssm_parameter" "environment_variables" {
+  count       = var.bootstrap ? 1 : 0
```
[the remark this answers is not in the document]
Comment (ben851) [reply aws/lambda-api/secrets_manager.tf:13]: No thanks

File: aws/lambda-api/secrets_manager.tf [hunk 7]
```diff
@@ -8,3 +8,122 @@
...
+
+FF_SALESFORCE_CONTACT=true
+FF_RTL=true
+FF_ANNUAL_LIMIT=true
+
+FIDO2_DOMAIN=${var.base_domain}
+FRESH_DESK_PRODUCT_ID=${var.manifest_fresh_desk_product_id}
+FRESH_DESK_API_URL=https://cds-snc.freshdesk.com
+FRESH_DESK_API_KEY=${var.manifest_fresh_desk_api_key}
+FRESH_DESK_ENABLED=False
```
[the remark this answers is not in the document]
Comment (ben851) [reply aws/lambda-api/secrets_manager.tf:66]: Agreed but this is what was in the file so I don't want to mess with it.

File: env/sandbox/lambda-api/terragrunt.hcl (hunk 1 above)
[the remark this answers is not in the document]
Comment (P0NDER0SA) [reply env/sandbox/lambda-api/terragrunt.hcl:68]: lol!

**Review by P0NDER0SA — COMMENTED** [review]
As we discussed we can probably handle these values with our configurations/tfvars -- but as a transitional measure I'm ok with this, and we can iterate on it as we mature our secrets management

**Review by P0NDER0SA — APPROVED** [review]
OK, i think this is ok for us to try

**Pull request merged**